rotkehlchen/chain/evm/transactions.py

```diff
@@ -554,3 +555,39 @@ def get_receipts_for_transactions_missing_them(
                     if 'UNIQUE constraint failed: evmtx_receipts.tx_hash' not in str(e):
                         log.error(f'Failed to store transaction {entry.hex()} receipt due to {str(e)}')  # noqa: E501
                         raise  # if receipt is already added by other greenlet it's fine
+
+    def add_transaction_hash(
+            self,
+            tx_hash: EVMTxHash,
+            associated_address: ChecksumEvmAddress,
+    ) -> tuple['EvmTransaction', 'EvmTxReceipt']:
+        """Adds a transaction to the database using its hash linking it to the provided address.
+
+        May raise:
+        - RemoteError if any of the remote queries fail.
+        - KeyError if there's a missing key in the tx_receipt dict.
+        - DeserializationError if there's an issue deserializing a value.
+        - pysqlcipher3.dbapi2.IntegrityError if the tx_hash is present in the db or address is not tracked.
+        """  # noqa: E501
+        dbevmtx = DBEvmTx(self.database)
+        transaction = self.evm_inquirer.get_transaction_by_hash(tx_hash)
+        receipt_data = self.evm_inquirer.get_transaction_receipt(tx_hash)
+        with self.database.user_write() as write_cursor:
+            dbevmtx.add_evm_transactions(
+                write_cursor=write_cursor,
+                evm_transactions=[transaction],
+                relevant_address=associated_address,
+            )
+            dbevmtx.add_receipt_data(
+                write_cursor=write_cursor,
+                chain_id=self.evm_inquirer.chain_id,
+                data=receipt_data,
+            )
+        with dbevmtx.db.conn.read_ctx() as cursor:
+            tx_receipt = dbevmtx.get_receipt(
+                cursor=cursor,
+                tx_hash=tx_hash,
+                chain_id=self.evm_inquirer.chain_id,
+            )
+        assert tx_receipt is not None, 'the transaction receipt is always present if no exception is raised'  # noqa: E501
```

Comment:
Hmm this assertion is weird here.

If indeed transaction receipt is always present when no exception is raised then the return type of `dbevmtx.get_receipt()` should not be optional.

But looking at the code in that function it seems to return `None` only if this happens:

```python
        chain_id_serialized = chain_id.serialize_for_db()
        cursor.execute(
            'SELECT contract_address, status, type from evmtx_receipts WHERE tx_hash=? AND chain_id=?',  # noqa: E501
            (tx_hash, chain_id_serialized))
        result = cursor.fetchone()
        if result is None:
            return None
```


Hm .. which seems just a silly way to say:

```python
        cursor.execute(
            'SELECT COUNT(*) from evmtx_receipts WHERE tx_hash=? AND chain_id=?',  # noqa: E501
            (tx_hash, chain_id_serialized))
        )
```

can you change this in that function? And maybe just rename the assertion a bit
```suggestion
        assert tx_receipt is not None, 'transaction receipt was added just above, so should exist'
```

Reply:
But the data from the `fetchone()` is used below in the function.

Reply:
It is? Oh my god lol brainfart. Then forget about it :rofl: 

Just change the assertion